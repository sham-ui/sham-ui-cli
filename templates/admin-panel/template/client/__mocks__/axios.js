const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
