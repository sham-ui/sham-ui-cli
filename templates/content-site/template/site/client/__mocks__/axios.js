const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
