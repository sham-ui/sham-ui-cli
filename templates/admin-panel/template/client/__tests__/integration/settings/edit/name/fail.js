import setup, { app } from '../../../helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
    setup();
} );

it( 'fail edit name', async() => {
    expect.assertions( 1 );

    axios.useDefaultMocks();

    history.pushState( {}, '', 'http://client.example.com/settings/' );
    await app.start();
    app.click( '.panel.settings p:nth-of-type(1) .icon-pencil' );

    const formData = {
        name: ''
    };
    axios
        .use( 'put', '/members/name', {
            'Status': 'Bad Name',
            'Messages': [ 'Name must have more than 0 characters.' ]
        }, 400 );

    app.form.fill( 'name', formData.name );
    await app.form.submit();

    app.checkBody();
} );
