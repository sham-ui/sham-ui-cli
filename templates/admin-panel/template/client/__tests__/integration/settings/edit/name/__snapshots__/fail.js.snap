// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fail edit name 1`] = `
"<div class=\\"\\">
  <div class=\\"logo\\">
    <a href=\\"\\"><span>[</span> {{ logoText }} <span>]</span></a>
  </div>
  <div class=\\"sideleft\\">
    <label class=\\"sidebar-label\\">Navigation</label>
    <div class=\\"menu\\">
      <a href=\\"\\">
        <div class=\\"menu-item\\">
          <i class=\\"icon-home-outline menu-item-icon\\"></i><span class=\\"menu-item-label\\">Dashboard</span>
        </div>
      </a>
      <!--if-->
    </div>
  </div>
  <div class=\\"header\\">
    <div class=\\"header-left\\">
      <div class=\\"navicon-left\\">
        <a>
          <i class=\\"icon-menu\\"></i>
        </a>
      </div>
    </div>
    <div class=\\"header-right\\">
      <div class=\\"dropdown\\">
        <a class=\\"link-profile\\"><span class=\\"logged-name\\">John Smith</span></a>
        <div class=\\"dropdown-menu \\">
          <ul class=\\"user-profile-nav\\">
            <li>
              <a href=\\"/settings\\">
                <i class=\\"icon-cog\\"></i> Settings
              </a>
            </li>
            <li>
              <a href=\\"#\\">
                <i class=\\"icon-logout\\"></i> Sign Out
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"mainpanel\\">
    <h4 class=\\"page-header\\"> Profile settings
      <!--header-->
    </h4>
    <div class=\\"page-body\\">
      <div class=\\"panel settings\\">
        <h6>Name</h6>
        <p class=\\"mg-b-30\\">John Smith<a>
            <i class=\\"icon-pencil\\"></i>
          </a>
          <form class=\\"form-layout\\">
            <!--header-->
            <ul class=\\"errors-panel\\">
              <li>Name must have more than 0 characters.</li>
            </ul>
            <!--if-->
            <!--ErrorsPanel-->
            <div class=\\"form-group\\">
              <label>Name:</label>
              <input name=\\"name\\" type=\\"text\\" autofocus=\\"\\">
            </div>
            <!--default-->
            <!--default-->
            <button type=\\"submit\\">
              <!--if-->Update
              <!--submit-->
            </button>
            <!--footer-->
          </form>
          <!--FormContainer-->
          <!--if-->
          <!--FormWithConfirmModal-->
          <!--DynamicComponent-->
          <!--if-->
        </p>
        <!--Paragraph-->
        <h6>Email</h6>
        <p class=\\"mg-b-30\\">j.smith@example.com<a>
            <i class=\\"icon-pencil\\"></i>
          </a>
          <!--if-->
        </p>
        <!--Paragraph-->
        <h6>Password</h6>
        <p class=\\"mg-b-0\\">******<a>
            <i class=\\"icon-pencil\\"></i>
          </a>
          <!--if-->
        </p>
        <!--Paragraph-->
      </div>
      <!--body-->
    </div>
    <!--if-->
    <!--Page-->
    <!--ActivePageContainer-->
  </div>
</div>
<!--LayoutAuthenticated-->
<!--if-->
<!--Layout-->
<!--if-->"
`;
